Lib/sqlite3/test/userfunctions.py

```diff
@@ -275,6 +276,20 @@ def CheckAnyArguments(self):
         val = cur.fetchone()[0]
         self.assertEqual(val, 2)
 
+    @unittest.skipIf(sqlite.sqlite_version_info < (3, 8, 3), "deterministic parameter not supported")
+    def CheckFuncDeterministic(self):
+        mock = unittest.mock.Mock()
+        mock.return_value = None
+        self.con.create_function("deterministic", 0, mock, deterministic=True)
+        cur = self.con.cursor()
+        cur.execute("select deterministic() = deterministic()")
+        self.assertEqual(mock.call_count, 1)
+
+    @unittest.skipIf(sqlite.sqlite_version_info >= (3, 8, 3), "SQLite < 3.8.3 needed")
+    def CheckFuncDeterministicNotSupported(self):
+        with self.assertRaises(sqlite.NotSupportedError):
+            self.con.create_function("deterministic", 0, int, deterministic=True)
```

Comment:
Could you also add a simple test to make sure that `deterministic` is a keyword-only argument?

```py
self.con.create_function("deterministic", 0, int, True)
```

Reply:
fixed in the following test